<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>947</width>
    <height>736</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Qt Modbus Tester</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,1">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayoutConnection">
      <item>
       <widget class="QLabel" name="labelConnection">
        <property name="text">
         <string>&lt;b&gt;Connection:&lt;/b&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelServer">
        <property name="text">
         <string>Server:</string>
        </property>
        <property name="buddy">
         <cstring>lineEditServer</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEditServer">
        <property name="text">
         <string>192.168.8.116</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelPort">
        <property name="text">
         <string>Port:</string>
        </property>
        <property name="buddy">
         <cstring>spinBoxPort</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxPort">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>502</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelSlave">
        <property name="text">
         <string>Slave:</string>
        </property>
        <property name="buddy">
         <cstring>spinBoxSlave</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxSlave">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonConnect">
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="labelConnectionStatus">
        <property name="text">
         <string notr="true">Connection status</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayoutErrorHandling">
      <item>
       <widget class="QLabel" name="labelErrorHandling">
        <property name="text">
         <string>&lt;b&gt;Error handling:&lt;/b&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelTimeout">
        <property name="text">
         <string>Timeout:</string>
        </property>
        <property name="buddy">
         <cstring>spinBoxTimeout</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxTimeout">
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelRetries">
        <property name="text">
         <string>Retries:</string>
        </property>
        <property name="buddy">
         <cstring>spinBoxRetries</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxRetries">
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxShowMsgBoxes">
        <property name="text">
         <string>Show messageboxes</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayoutDataUnit">
      <item>
       <widget class="QLabel" name="labelDataUnit">
        <property name="text">
         <string>&lt;b&gt;Data unit:&lt;/b&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelType">
        <property name="text">
         <string>Type:</string>
        </property>
        <property name="buddy">
         <cstring>comboBoxType</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboBoxType"/>
      </item>
      <item>
       <widget class="QLabel" name="labelRegister">
        <property name="text">
         <string>Register:</string>
        </property>
        <property name="buddy">
         <cstring>spinBoxRegister</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxRegister">
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelCount">
        <property name="text">
         <string>Count:</string>
        </property>
        <property name="buddy">
         <cstring>spinBoxCount</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxCount">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxAutorefresh">
        <property name="text">
         <string>Autorefresh</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelDelay">
        <property name="text">
         <string>Delay:</string>
        </property>
        <property name="buddy">
         <cstring>spinBoxDelay</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxDelay">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="suffix">
         <string>ms</string>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonRequest">
        <property name="text">
         <string>Request</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonWrite">
        <property name="text">
         <string>Write values</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="labelRequestStatus">
        <property name="text">
         <string>Idle</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTableView" name="tableView"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>947</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>checkBoxAutorefresh</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinBoxDelay</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>518</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>630</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
